components/hab/src/cli.rs

```diff
@@ -1316,6 +1324,47 @@ fn non_empty(val: String) -> result::Result<(), String> {
     }
 }
 
+/// Adds extra configuration options for shutting down a service with
+/// a customized shutdown signal and timeout.
+///
+/// These are currently feature-flagged. Eventually, we hope to have
+/// this be definable in a package or at load time.
+#[cfg(unix)]
+fn maybe_add_configurable_shutdown_options(mut app: App<'static, 'static>,
+                                           feature_flags: FeatureFlag)
+                                           -> App<'static, 'static> {
+    if feature_flags.contains(FeatureFlag::CONFIGURE_SHUTDOWN) {
+        app = app.arg(Arg::with_name("SHUTDOWN_SIGNAL").help("The signal to send to a service \
+                                                              to safely shut it down")
+                                                       .long("shutdown-signal")
+                                                       .takes_value(true)
+                                                       .default_value(&SHUTDOWN_SIGNAL_DEFAULT));
+        app = add_shutdown_timeout_option(app);
+    }
+
+    app
+}
+
+#[cfg(windows)]
+fn maybe_add_configurable_shutdown_options(mut app: App<'static, 'static>,
+                                           feature_flags: FeatureFlag)
+                                           -> App<'static, 'static> {
+    // Identical to the Unix implementation, but does not include the
+    // SHUTDOWN_SIGNAL option, since that doesn't exist on Windows
```

Comment:
Hrmm, good catch. I'll need to expose the signal types and defaults for Windows, but it would only get used on the CLI side; all the Supervisor-side code can stay the same.